feat: Improve Fuel Codes and Tables

dhaselhan opened the pull request:
* Fix saving issues with Fuel Code and Fuel Code Other
* Update Fuel codes code to be a suffix, since the final fuel code is calculated
*Add a fuel code that can be used for display
* Use new display property to align dropdown options and values
* Better names for Schema validation for Fuel Supply
* Add missing required stars to Add Fuel Code table
* Clean up and refactor row error validation to a more standard format
* Add logic to handle and display warnings

Closes: #864

AlexZorkin reviewed (changes requested):
I added a few fixes on the details object and various schedules. There's one more bug we have to fix for allocation agreements, where the fuel_code field is sending back the prefix+suffix, but is then breaking when it looks up the fuel_code by suffix only.

/backend/lcfs/web/api/allocation_agreement/services.py
        fuel_code = await self.fuel_repo.get_fuel_code_by_name(
            allocation_agreement.fuel_code
        )
We will have to parse out the suffix from the prefix+suffix to lookup by name.

Otherwise this is looking done. ty

dhaselhan commented:
> I added a few fixes on the details object and various schedules. There's one more bug we have to fix for allocation agreements, where the fuel_code field is sending back the prefix+suffix, but is then breaking when it looks up the fuel_code by suffix only.
> 
> /backend/lcfs/web/api/allocation_agreement/services.py fuel_code = await self.fuel_repo.get_fuel_code_by_name( allocation_agreement.fuel_code ) We will have to parse out the suffix from the prefix+suffix to lookup by name.
> 
> Otherwise this is looking done. ty

Got the query updated and fixed some issues with the errors fields.

AlexZorkin reviewed (approved):
Thank you for the updates. 👍
